fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MaterializedViewRewriter.java

```diff
@@ -1013,25 +1020,56 @@ private List<BiMap<Integer, Integer>> generateRelationIdMap(
         List<BiMap<Integer, Integer>> result = ImmutableList.of(HashBiMap.create());
         for (Map.Entry<Table, Set<Integer>> queryEntry : queryTableToRelationId.entrySet()) {
             Preconditions.checkState(!queryEntry.getValue().isEmpty());
-            if (queryEntry.getValue().size() == 1) {
-                Integer src = queryEntry.getValue().iterator().next();
-                Integer target = mvTableToRelationId.get(queryEntry.getKey()).iterator().next();
-                for (BiMap<Integer, Integer> m : result) {
-                    m.put(src, target);
+            Set<Integer> queryTableIds = queryEntry.getValue();
+            Set<Integer> mvTableIds = mvTableToRelationId.get(queryEntry.getKey());
+            if (queryTableIds.size() == 1) {
```

Comment:
Query table's size is one & MV's table size is one, the mapping is fixed, just add the mapping to the BiMap and no need more additional mappings.
otherwise need additional mappings to handle Query table id mapings with MV's table ids.

like:
case1: will add one more additional mapping
// Query table ids: a
// MV    table ids: a1, a2

case2: will add one more additional mapping
 // Query table ids: a1, a2
  // MV    table ids: A1, A2
  // output:
  //  a1 -> A1, a2 -> A2
  //  a1 -> A2, a2 -> A1


BTW, This changes come from TPCH query22 which query contains multi same tables and have different names with MVs.
```
select
    s_name,
    count(*) as numwait
from
    supplier,
    lineitem l1,
    orders,
    nation
where
        s_suppkey = l1.l_suppkey
  and o_orderkey = l1.l_orderkey
  and o_orderstatus = 'F'
  and l1.l_receiptdate > l1.l_commitdate
  and exists (
        select
            *
        from
            lineitem l2
        where
                l2.l_orderkey = l1.l_orderkey
          and l2.l_suppkey <> l1.l_suppkey
    )
  and not exists (
        select
            *
        from
            lineitem l3
        where
                l3.l_orderkey = l1.l_orderkey
          and l3.l_suppkey <> l1.l_suppkey
          and l3.l_receiptdate > l3.l_commitdate
    )
  and s_nationkey = n_nationkey
  and n_name = 'CANADA'
group by
    s_name
order by
    numwait desc,
    s_name limit 100;
```